src/legacy/ui/public/agg_types/controls/field.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { isFunction } from 'lodash';
+import React, { useEffect } from 'react';
+
+import { EuiComboBox, EuiComboBoxOptionProps, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { get } from 'lodash';
+import { AggConfig } from 'ui/vis';
+import { formatListAsProse, parseCommaSeparatedList } from '../../../../utils';
+import { AggParamEditorProps } from '../../vis/editors/default';
+import { ComboBoxGroupedOption } from '../../vis/editors/default/default_editor_utils';
+import { FieldParamType } from '../param_types';
+
+function FieldParamEditor({
+  agg = {},
+  indexedFields = [],
+  aggParam,
+  value,
+  setValue,
+  isInvalid,
+  setValidity,
+  setTouched,
+}: AggParamEditorProps<FieldParamType>) {
+  const label = i18n.translate('common.ui.aggTypes.field.fieldLabel', { defaultMessage: 'Field' });
+  const selectedOptions: ComboBoxGroupedOption[] = value
+    ? [{ label: value.displayName, value }]
+    : [];
+
+  const onChange = (options: EuiComboBoxOptionProps[]) => {
+    setValue(get(options, '0.value'));
+
+    if (isFunction(aggParam.onChange)) {
+      aggParam.onChange(agg);
+    }
+  };
+  const errors = [];
+
+  if (!indexedFields.length) {
+    setTouched();
+    errors.push(
+      i18n.translate('common.ui.aggTypes.field.noCompatibleFieldsDescription', {
+        defaultMessage:
+          'The {indexPatternTitle} index pattern does not contain any of the following compatible field types: {fieldTypes}',
+        values: {
+          indexPatternTitle: agg.getIndexPattern && agg.getIndexPattern().title,
+          fieldTypes: getFieldTypesString(agg),
+        },
+      })
+    );
+  }
+
+  useEffect(
+    () => {
+      setValidity(!!value);
+    },
+    [value]
+  );
+
+  return (
+    <EuiFormRow
+      label={label}
+      isInvalid={isInvalid}
+      fullWidth={true}
+      error={errors}
+      helpText={
+        <FormattedMessage
+          id="inputControl.editor.listControl.parentDescription"
+          defaultMessage="Options are based on the index pattern."
```

Comment:
I think this works great! But do we think we need this normal help text?

Reply:
I agree, we don't need it.